types/stampit/index.d.ts

```diff
@@ -1,518 +1,560 @@
-// Type definitions for stampit 3.0
+// Type definitions for stampit 4.3
 // Project: https://github.com/stampit-org/stampit, https://stampit.js.org
 // Definitions by: Vasyl Boroviak <https://github.com/koresar>
 //                 Harris Lummis <https://github.com/lummish>
+//                 PopGoesTheWza <https://github.com/popgoesthewza>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.5
+
+// Utility types
+
+/**
+ * @internal Base type for all `methods`-like metadata.
+ * @template This The type to use for `this` within methods.
+ */
+interface MethodMap<This> {
+    [s: string]: (this: This, ...args: any[]) => any;
+}
+
+/** @internal A plain old JavaScript object created by a `Stamp`. */
+type Pojo = object; // { [s: string]: any; }
+
+/** @internal Base type for all `properties`-like metadata. */
+// TODO: discriminate Array
+type PropertyMap = object; // { [s: string]: any; }
+
+/** @internal Signature common to every `Stamp`s. */
+interface StampSignature {
+    (options?: PropertyMap, ...args: unknown[]): any;
+    compose: ComposeMethod & stampit.Descriptor<any, any>;
```

Comment:
After renaming `stampit.Descriptor` to `stampit.ExtendedDescriptor` we should add (somewhere above) the `Descriptor`. It should have only these exactly 11 properties: https://github.com/stampit-org/stamp-specification#the-stamp-descriptor

Reply:
In other words, at runtime the `MyStamp.compose.*` properties are limited to those 11. There will never be `props`, or `init`, or other shortened properties.

Reply:
```
compose: ComposeMethod & Descriptor;
```

Reply:
Ok

Reply:
Done